internal/cli/status.go

```diff
@@ -0,0 +1,850 @@
+		}
+	}
+
+	return 0
+}
+
+// FormatProjectAppStatus formats all applications inside a project
+func (c *StatusCommand) FormatProjectAppStatus(projectTarget string) error {
+	if !c.flagJson {
+		c.ui.Output(wpStatusProjectMsg, projectTarget, c.serverCtx.Server.Address)
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	resp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: projectTarget,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	project := resp.Project
+
+	workspace, err := c.getWorkspaceFromProject(resp)
+	if err != nil {
+		return err
+	}
+
+	// Summary
+	//   App list
+
+	appHeaders := []string{
+		"App", "Workspace", "Latest Status", "Last Check",
+	}
+
+	appTbl := terminal.NewTable(appHeaders...)
+
+	appFailures := false
+	for _, app := range project.Applications {
+		appStatusResp, err := client.GetLatestStatusReport(c.Ctx, &pb.GetLatestStatusReportRequest{
+			Application: &pb.Ref_Application{
+				Application: app.Name,
+				Project:     project.Name,
+			},
+			Workspace: &pb.Ref_Workspace{
+				Workspace: workspace,
+			},
+		})
+		if status.Code(err) == codes.NotFound {
+			// App doesn't have a status report yet, likely not deployed
+			err = nil
+		}
+		if err != nil {
+			return err
+		}
+
+		statusReportComplete, statusReportCheckTime, err := c.FormatStatusReportComplete(appStatusResp)
+		if err != nil {
+			return err
+		}
+
+		statusColor := ""
+		columns := []string{
+			app.Name,
+			workspace,
+			statusReportComplete,
+			statusReportCheckTime,
+		}
+
+		// Add column data to table
+		appTbl.Rich(
+			columns,
+			[]string{
+				statusColor,
+			},
+		)
+	}
+
+	if c.flagJson {
+		c.outputJsonProjectAppStatus(appTbl, project)
+	} else {
+		c.ui.Output("")
+		c.ui.Table(appTbl, terminal.WithStyle("Simple"))
+		c.ui.Output("")
+		c.ui.Output(wpStatusProjectSuccessMsg)
+	}
+
+	if appFailures {
+		c.ui.Output("")
+
+		c.ui.Output(wpStatusHealthTriageMsg, projectTarget, terminal.WithWarningStyle())
+	}
+
+	return nil
+}
+
+func (c *StatusCommand) FormatAppStatus(projectTarget string, appTarget string) error {
+	if !c.flagJson {
+		c.ui.Output(wpStatusAppProjectMsg, appTarget, projectTarget, c.serverCtx.Server.Address)
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	projResp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: projectTarget,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	project := projResp.Project
+
+	workspace, err := c.getWorkspaceFromProject(projResp)
+	if err != nil {
+		return err
+	}
+
+	// App Summary
+	//  Summary of single app
+	var app *pb.Application
+	for _, a := range project.Applications {
+		if a.Name == appTarget {
+			app = a
+			break
+		}
+	}
+	if app == nil {
+		return fmt.Errorf(fmt.Sprintf("Did not find aplication %q in project %q", appTarget, projectTarget))
+	}
+
+	appStatusResp, err := client.GetLatestStatusReport(c.Ctx, &pb.GetLatestStatusReportRequest{
```

Comment:
Same problem here - I think you need more than just the most recent deploy or release status report to determine overall app status.

One piece of WP-075 is building APIs that get you a deployment/release with it's corresponding latest status report, which I think would be useful here. I've been holding off building that until WP-080 stabilizes, but I think it's just what you need here.

Reply:
Since this is the most detailed view, I could basically grab the status report associated with the latest deployment and or release (if a release happened) and show them both here in separate columns.